### PR TITLE
Validate versions in metadata.json files are sorted

### PR DESCRIPTION
**Pull request by meteorcloudy**
Addressing https://github.com/bazelbuild/bazel-central-registry/pull/1425#issuecomment-1925761218

### CODE REVIEW

File: modules/rules_xcodeproj/metadata.json [hunk 1]
```diff
@@ -16,13 +16,13 @@
...
     ],
     "versions": [
-        "1.0.0rc1",
-        "1.0.0rc2",
-        "1.0.0rc3",
         "0.12.0",
         "0.12.2",
         "0.12.3",
         "1.0.1",
+        "1.0.0rc1",
```
Comment (meteorcloudy) [inline modules/rules_xcodeproj/metadata.json:23]: Unfortunately, our current version format will actually put `1.0.0rc1` after `1.0.1`, `1.0.0-rc1` would have worked correctly though.

File: modules/buildozer/metadata.json [hunk 2]
```diff
@@ -11,8 +11,8 @@
         "github:fmeum/buildozer"
     ],
     "versions": [
-        "6.4.0.2",
-        "6.4.0"
+        "6.4.0",
+        "6.4.0.2"
```
[the remark this answers is not in the document]
Comment (kormide) [reply modules/buildozer/metadata.json:15]: The algorithm is that it sorts non-semvers lexicographically and puts them at the top of the list, then it sorts semvers by version and puts them at the bottom. I don't think `6.4.0.2` is a valid semver, is it?

[the remark this answers is not in the document]
Comment (kormide) [reply modules/buildozer/metadata.json:15]: In any case, it should match the algorithm that's in the BCR..

[the remark this answers is not in the document]
Comment (Wyverald) [reply modules/buildozer/metadata.json:15]: see https://github.com/bazelbuild/bazel-central-registry/blob/127d91703baf4e39eb66fc907d255b37d6162792/tools/registry.py#L82-L128 for the sorting algorithm for bzlmod versions

[the remark this answers is not in the document]
Comment (Wyverald) [reply modules/buildozer/metadata.json:15]: and https://bazel.build/external/module#version_format for an overview of the format

[the remark this answers is not in the document]
Comment (kormide) [reply modules/buildozer/metadata.json:15]: fyi: https://github.com/bazel-contrib/publish-to-bcr/pull/145

File: modules/rules_xcodeproj/metadata.json (hunk 1 above)
Comment (brentleyjones) [reply modules/rules_xcodeproj/metadata.json:23]: Yeah, it was that way, which is why I had to release 1.0.1.

**Pull request merged**